consensus/utils.go

```diff
@@ -6,9 +6,51 @@ import (
 
 	"github.com/deso-protocol/core/bls"
 	"github.com/deso-protocol/core/collections"
+	"github.com/holiman/uint256"
 	"golang.org/x/crypto/sha3"
 )
 
+// Given a QC and a sorted validator set, this function returns true if the QC contains a valid
+// super-majority of signatures from the validator set for the QC's (View, BlockHash) pair.
+func IsValidSuperMajorityQuorumCertificate(qc QuorumCertificate, validators []Validator) bool {
+	if !isProperlyFormedQC(qc) || !isProperlyFormedValidatorSet(validators) {
+		return false
+	}
+
+	// Compute the signature that validators in the QC would have signed
+	signaturePayload := GetVoteSignaturePayload(qc.GetView(), qc.GetBlockHash())
+
+	// Compute the total stake in the QC and the total stake in the network
+	stakeInQC := uint256.NewInt()
+	totalStake := uint256.NewInt()
+
+	// Fetch the validators in the QC
+	validatorPublicKeysInQC := []*bls.PublicKey{}
+
+	// Fetch the validators in the QC, and compute the sum of stake in the QC and in the network
+	for ii := range validators {
```

Comment:
Should we also validate that `len(validators) == qc.GetSignersList().Size()`?

Reply:
No actually. The Bigint.ToBytes function compresses the byte encoding for a BigInt such if there's a sequence of 0 at higher indices, of the construct, then they are removed from the byte encoding and always assumed to be zero. This way, `len(validators) >= qc.GetSignersList().Size()` for Indices that exist in the validators slice but not in the signers list return a false value.

Ex: if we have a the sequence `true, true, true, false true, false, false, false, false, false`, the binary representation is `0000010111`, which gets compressed to `10111`.  Calling `bitset.Get(8)` returns a value of false for index 9.